component/src/main/java/org/wso2/carbon/identity/authenticator/office365/Office365Authenticator.java

```diff
@@ -0,0 +1,339 @@
+    }
+
+    /**
+     * Get the Context identifier sent with the request.
+     */
+    @Override
+    public String getContextIdentifier(HttpServletRequest request) {
+        return request.getParameter(Office365AuthenticatorConstants.STATE);
+    }
+
+    /**
+     * Check whether the authentication or logout request can be handled by the authenticator
+     */
+    @Override
+    public boolean canHandle(HttpServletRequest arg0) {
+        return true;
+    }
+
+    /**
+     * Authenticator flow process
+     */
+    @Override
+    public AuthenticatorFlowStatus process(HttpServletRequest request, HttpServletResponse response,
+                                           AuthenticationContext context)
+            throws AuthenticationFailedException, LogoutFailedException {
+        // if the logout request comes, then no need to go through and doing complete the flow.
+        if (context.isLogoutRequest()) {
+            return AuthenticatorFlowStatus.SUCCESS_COMPLETED;
+        }
+        if (StringUtils.isNotEmpty(request.getParameter(Office365AuthenticatorConstants.CODE))) {
+            processAuthenticationResponse(request, response, context);
+            return AuthenticatorFlowStatus.SUCCESS_COMPLETED;
+        } else {
+            initiateAuthenticationRequest(request, response, context);
+        }
+        return AuthenticatorFlowStatus.INCOMPLETE;
+    }
+
+    /**
+     * initiate the authentication request
+     */
+    @Override
+    protected void initiateAuthenticationRequest(HttpServletRequest request,
+                                                 HttpServletResponse response, AuthenticationContext context)
+            throws AuthenticationFailedException {
+        Map<String, String> authenticatorProperties = context.getAuthenticatorProperties();
+        String clientId = authenticatorProperties.get(OIDCAuthenticatorConstants.CLIENT_ID);
+        String redirectUri = authenticatorProperties.get(Office365AuthenticatorConstants.CALLBACK_URL);
+        String loginPage = getAuthorizationServerEndpoint(context.getAuthenticatorProperties());
+        String queryParams = Office365AuthenticatorConstants.STATE + "=" + context.getContextIdentifier();
+        try {
+            response.sendRedirect(response.encodeRedirectURL(loginPage + "?" + queryParams + "&" +
+                    Office365AuthenticatorConstants.CLIENT_ID + "=" + clientId + "&" +
+                    Office365AuthenticatorConstants.RESPONSE_TYPE + "=" +
+                    OIDCAuthenticatorConstants.OAUTH2_GRANT_TYPE_CODE + "&" +
+                    Office365AuthenticatorConstants.REDIRECT_URI + "=" + redirectUri + "&" +
+                    Office365AuthenticatorConstants.Resource + "=" + Office365AuthenticatorConstants.OFFICE365_RESOURCE));
+        } catch (IOException e) {
+            throw new AuthenticationFailedException("Error while redirecting", e);
+        }
+    }
+
+    /**
+     * Process the response of the office365 end-point
```

Comment:
Javadocs need to mention about the request params and the response. As mention above, the first line should end with a full stop, else when building in Windows this will fail.

Reply:
Fixed in https://github.com/wso2-extensions/identity-outbound-auth-office365/pull/3